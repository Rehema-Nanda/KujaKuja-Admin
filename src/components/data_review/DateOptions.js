export default class DateOptions {
    static get CREATED_AT() {
        return "Created At";
    }

    static get UPLOADED_AT() {
        return "Uploaded At";
    }

    static get ALL_OPTIONS() {
        return [this.CREATED_AT, this.UPLOADED_AT];
    }
}
