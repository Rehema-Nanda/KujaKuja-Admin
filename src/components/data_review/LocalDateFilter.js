import React from "react";
import {
    Form, FormGroup, Label, Input,
} from "reactstrap";

import DateOptions from "./DateOptions";

const LocalDateFilter = ({ handleLocalFilterDateOptionSelect, selectedDateOption }) => {
    return (
        <Form className="local-date-container">
            <FormGroup>
                {DateOptions.ALL_OPTIONS.map((option) => {
                    return (
                        <FormGroup check key={option} inline>
                            <Label check className="form-label-check">
                                <Input
                                    type="radio"
                                    name="radio1"
                                    onChange={handleLocalFilterDateOptionSelect}
                                    value={option}
                                    checked={option === selectedDateOption}
                                />
                                {option}
                            </Label>
                        </FormGroup>
                    );
                })}
            </FormGroup>
        </Form>
    );
};

export default LocalDateFilter;
