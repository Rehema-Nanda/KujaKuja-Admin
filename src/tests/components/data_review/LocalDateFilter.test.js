import React from "react";
import { mount } from "enzyme";

import LocalDateFilter from "../../../components/data_review/LocalDateFilter";
import DateOptions from "../../../components/data_review/DateOptions";

describe("LocalDateFilter component", () => {
    const props = {
        handleLocalFilterDateOptionSelect: jest.fn(),
        selectedDateOption: DateOptions.CREATED_AT,
    };

    const wrapper = mount(<LocalDateFilter {...props} />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
