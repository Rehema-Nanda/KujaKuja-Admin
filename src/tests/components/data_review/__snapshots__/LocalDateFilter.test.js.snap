// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalDateFilter component renders correctly 1`] = `
"<LocalDateFilter handleLocalFilterDateOptionSelect={[Function: mockConstructor]} selectedDateOption=\\"Created At\\">
  <Form className=\\"local-date-container\\" tag=\\"form\\">
    <form className=\\"local-date-container\\">
      <FormGroup tag=\\"div\\">
        <div className=\\"position-relative form-group\\">
          <FormGroup check={true} inline={true} tag=\\"div\\">
            <div className=\\"position-relative form-check form-check-inline\\">
              <Label check={true} className=\\"form-label-check\\" tag=\\"label\\" widths={{...}}>
                <label htmlFor={[undefined]} className=\\"form-label-check form-check-label\\">
                  <Input type=\\"radio\\" name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Created At\\" checked={true}>
                    <input name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Created At\\" checked={true} type=\\"radio\\" className=\\"form-check-input\\" />
                  </Input>
                  Created At
                </label>
              </Label>
            </div>
          </FormGroup>
          <FormGroup check={true} inline={true} tag=\\"div\\">
            <div className=\\"position-relative form-check form-check-inline\\">
              <Label check={true} className=\\"form-label-check\\" tag=\\"label\\" widths={{...}}>
                <label htmlFor={[undefined]} className=\\"form-label-check form-check-label\\">
                  <Input type=\\"radio\\" name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Uploaded At\\" checked={false}>
                    <input name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Uploaded At\\" checked={false} type=\\"radio\\" className=\\"form-check-input\\" />
                  </Input>
                  Uploaded At
                </label>
              </Label>
            </div>
          </FormGroup>
        </div>
      </FormGroup>
    </form>
  </Form>
</LocalDateFilter>"
`;
